module/filter.mjs

```diff
@@ -0,0 +1,362 @@
+
+/**
+ * Perform an NAND check against all filters.
+ * @param {object} data                 Data to check.
+ * @param {FilterDescription[]} filter  Filter to compare against.
+ * @returns {boolean}
+ */
+export function NAND(data, filter) {
+  return !filter.every(({k, v, o}) => _check(data, k, v, o));
+}
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Perform an OR check against all filters.
+ * @param {object} data                 Data to check.
+ * @param {FilterDescription[]} filter  Filter to compare against.
+ * @returns {boolean}
+ */
+export function OR(data, filter) {
+  return filter.some(({k, v, o}) => _check(data, k, v, o));
+}
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Perform an NOR check against all filters.
+ * @param {object} data                 Data to check.
+ * @param {FilterDescription[]} filter  Filter to compare against.
+ * @returns {boolean}
+ */
+export function NOR(data, filter) {
+  return !filter.some(({k, v, o}) => _check(data, k, v, o));
+}
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Perform an XOR check against all filters.
+ * @param {object} data                 Data to check.
+ * @param {FilterDescription[]} filter  Filter to compare against.
+ * @returns {boolean}
+ */
+export function XOR(data, filter) {
+  let currentResult = false;
+  for ( const { k, v, o } of filter ) {
+    if ( _check(data, k, v, o) ) {
+      if ( !currentResult ) currentResult = true;
+      else return false;
+    }
+  }
+  return currentResult;
+}
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Invert the result of a nested check,
+ * @param {object} data               Data to check.
+ * @param {FilterDescription} filter  Filter to compare against.
+ * @returns {boolean}
+ */
+export function NOT(data, filter) {
+  const { k, v, o } = filter;
+  return !_check(data, k, v, o);
+}
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+/*                  Comparison Functions                 */
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Currently supported comparison functions.
+ * @enum {Function}
+ */
+export const COMPARISON_FUNCTIONS = {
+  _: exact, exact, iexact, contains, icontains,
+  startswith, istartswith, endswith, iendswith,
+  has, hasAny, hasAll, in: in_, gt, gte, lt, lte
+};
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Check for an exact match. The default comparison mode if none is provided.
+ * @param {*} data
+ * @param {*} value
+ * @returns {boolean}
+ */
+export function exact(data, value) {
+  return data === value;
+}
+
+/* <><><><> <><><><> <><><><> <><><><> <><><><> <><><><> */
+
+/**
+ * Case-insensitive exact match.
+ * @param {*} data
+ * @param {*} value
+ * @returns {boolean}
+ */
+export function iexact(data, value) {
+  return exact(String(data).toLowerCase(), String(value).toLowerCase());
```

Comment:
Since we aren't using any case insensitive filters at the moment, maybe I'll just pull them out for the moment and we can stick them back if we find a need for them and make the decision then.